turbo/app/reset-datadir.go

```diff
@@ -0,0 +1,216 @@
+	}
+)
+
+func resetCliAction(cliCtx *cli.Context) (err error) {
+	// Set logging verbosity. Oof that function signature.
+	logger, _, _, _, err := debug.Setup(cliCtx, true)
+	if err != nil {
+		err = fmt.Errorf("setting up logging: %w", err)
+		return
+	}
+	prune := pruneFlag.Get(cliCtx)
+	dryRun := dryRunFlag.Get(cliCtx)
+	dataDirPath := cliCtx.String(utils.DataDirFlag.Name)
+
+	configChainName, err := getChainNameFromChainData(cliCtx, logger)
+	if err != nil {
+		return fmt.Errorf("getting chain name from chaindata: %w", err)
+	}
+	chain := utils.ChainFlag.Get(cliCtx)
+	if cliCtx.IsSet(utils.ChainFlag.Name) {
+		if configChainName.Ok && configChainName.Value != chain {
+			// Pedantic but interesting.
+			logger.Warn("chain name flag and chain config do not match", "flag", chain, "config", configChainName.Value)
+		}
+		logger.Info("using chain name from flag", "chain", chain)
+	} else {
+		if !configChainName.Ok {
+			return errors.New("chain flag not set and chain name not found in chaindata (reset already occurred or invalid data dir?)")
+		}
+		chain = configChainName.Unwrap()
+		logger.Info("read chain name from config", "chain", chain)
+	}
+
+	dirs := datadir.Open(dataDirPath)
+	unlock, err := dirs.TryFlock()
+	if err != nil {
+		return fmt.Errorf("failed to lock data dir %v: %w", dirs.DataDir, err)
+	}
+	defer unlock()
+	err = snapcfg.LoadRemotePreverified(cliCtx.Context)
+	if err != nil {
+		// TODO: Check if we should continue? What if we ask for a git revision and
+		// can't get it? What about a branch? Can we reset to the embedded snapshot hashes?
+		return fmt.Errorf("loading remote preverified snapshots: %w", err)
+	}
+	cfg := snapcfg.KnownCfg(chain)
+	// Should we check cfg.Local? We could be resetting to the preverified.toml...?
+	logger.Info(
+		"Loaded preverified snapshots hashes",
+		"len", len(cfg.Preverified.Items),
+		"chain", chain,
+	)
+	removeFunc := func(path string) error {
+		return os.Remove(filepath.Join(dirs.Snap, path))
+	}
+	if dryRun {
+		removeFunc = dryRunRemove
+	}
+	reset := reset{prune: prune}
+	logger.Info("Walking snapshots directory", "path", dirs.Snap)
+	err = reset.walkSnapshots(logger, dirs.Snap, cfg.Preverified, removeFunc)
+	if err != nil {
+		err = fmt.Errorf("walking snapshots: %w", err)
+		return
+	}
+	// Remove chaindata last, so that the config is available if there's an error.
+	logger.Warn("Removing chaindata dir", "path", dirs.Chaindata)
+	if !dryRun {
+		err = os.RemoveAll(dirs.Chaindata)
+	}
+	if err != nil {
+		err = fmt.Errorf("removing chaindata dir: %w", err)
+		return
+	}
+	return
+}
+
+func getChainNameFromChainData(cliCtx *cli.Context, logger log.Logger) (_ g.Option[string], err error) {
+	ctx := cliCtx.Context
+	nodeConfig, err := NewNodeConfig(cliCtx, logger)
+	if err != nil {
+		err = fmt.Errorf("getting node config: %w", err)
+		return
+	}
+	// Why does this fail if we set readonly?
+	db, err := node.OpenDatabase(cliCtx.Context, nodeConfig, kv.ChainDB, "", false, logger)
```

Comment:
do
```
db, err := mdbx.New(kv.ChainDB, log.New()).
		Path(dirs.Chaindata).
		Accede(true).
		Open(ctx)
```

then you will not need any db configs/params/flags and it will return error if db doesn't exists (instead of creating it)

Reply:
This worked, thanks.